### PR TITLE
SDK-304 [Java SDK] Allow oAuth2 token Storage initialization

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-304](https://incountry.atlassian.net/browse/SDK-304)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,6 +157,24 @@ StorageConfig config = new StorageConfig()
 Storage storage = StorageImpl.newStorage(config);
 ```
 
+The SDK also allows using previously acquired oAuth tokens if needed or pass token getting function. In this mode SDK is not responsible for oAuth a token renewal, and it should be done by SDK user himself.
```
Comment (RommeroInCountry) [inline README.md:160]: The SDK also allows using previously acquired oAuth token if needed or pass token accessor function

Comment (RommeroInCountry) [reply README.md:160]: In this mode SDK is not responsible for oAuth token ... (without "a")

Comment (RommeroInCountry) [reply README.md:160]: ...and it should be handled by SDK user himself. (I know we used done in node but "handle" is much better here

File: src/main/java/com/incountry/residence/sdk/StorageImpl.java [hunk 2]
```diff
@@ -44,7 +45,7 @@
     private static final ValidationHelper HELPER = new ValidationHelper(LOG);
     //error messages
     private static final String MSG_ERR_PASS_ENV = "Please pass environment_id param or set INC_ENVIRONMENT_ID env var";
-    private static final String MSG_ERR_AUTH_DUPL = "Either apiKey or clientId/clientSecret can be used at the same moment, not both";
+    private static final String MSG_ERR_AUTH = "Please pass only one parameter combination for authorization: clientId/clientSecret or apiKey or oauthTokenAccessor";
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/StorageImpl.java:48]: Please use only one auth authorization: clientId/clientSecret or apiKey or oauthTokenAccessor

**Review by RommeroInCountry — APPROVED** [review]

**Review by robert-par — APPROVED** [review]

File: src/main/java/com/incountry/residence/sdk/oauth/OauthTokenAccessor.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.incountry.residence.sdk.oauth;
+
+/**
+ * OAuth2 token accessor. Method {@link OauthTokenAccessor#getToken()} is invoked on HTTP interaction with InCountry storage service.
+ */
+public interface OauthTokenAccessor {
```
Comment (robert-par) [inline src/main/java/com/incountry/residence/sdk/oauth/OauthTokenAccessor.java:6]: Please mark OauthTokenAccessor with @FunctionalInterface annotation

File: README.md (hunk 1 above)
Comment (VladimirLevitskiy) [reply README.md:160]: > by SDK users themselves

**Pull request merged**